### PR TITLE
New environment fixes for cloudwatch enable

### PR DESCRIPTION
**Pull request by ben851**
# Summary | Résumé

There were some issues with enabling cloudwatch on dev - this should resolve them.

The import and removed blocks should be removed in a second PR once this is released.

Also sync'ing the dev lockfiles.

## Related Issues | Cartes liées

* https://app.zenhub.com/workspaces/notify-planning-core-6411dfb7c95fb80014e0cab0/issues/gh/cds-snc/notification-planning-core/467

## Before merging this PR

Read code suggestions left by the
[cds-ai-codereviewer](https://github.com/cds-snc/cds-ai-codereviewer/) bot. Address
valid suggestions and shortly write down reasons to not address others. To help
with the classification of the comments, please use these reactions on each of the
comments made by the AI review:

| Classification      | Reaction | Emoticon |
|---------------------|----------|----------|
| Useful              | +1       | 👍        |
| Noisy               | eyes     | 👀        |
| Hallucination       | confused | 😕        |
| Wrong but teachable | rocket   | 🚀        |
| Wrong and incorrect | -1       | 👎        |

The classifications will be extracted and summarized into an analysis of how helpful
or not the AI code review really is.

## Test instructions | Instructions pour tester la modification

TF Apply works and nothing breaks.

## Release Instructions | Instructions pour le déploiement

None.

## Reviewer checklist | Liste de vérification du réviseur

* [ ] This PR does not break existing functionality.
* [ ] This PR does not violate GCNotify's privacy policies.
* [ ] This PR does not raise new security concerns. Refer to our GC Notify Risk Register document on our Google drive.
* [ ] This PR does not significantly alter performance.
* [ ] Additional required documentation resulting of these changes is covered (such as the README, setup instructions, a related ADR or the technical documentation).

> ⚠ If boxes cannot be checked off before merging the PR, they should be moved to the "Release Instructions" section with appropriate steps required to verify before release. For example, changes to celery code may require tests on staging to verify that performance has not been affected.

### CODE REVIEW

File: aws/database-tools/blazer.tf [hunk 1]
```diff
@@ -45,7 +45,7 @@ resource "aws_ecs_task_definition" "blazer" {
       "logConfiguration" : {
         "logDriver" : "awslogs",
         "options" : {
-          "awslogs-group" : "${var.cloudwatch_enabled ? aws_cloudwatch_log_group.blazer[0].name : "none"}",
+          "awslogs-group" : "${var.cloudwatch_enabled ? "blazer" : "none"}",
```
[the remark this answers is not in the document]
Comment (ben851) [reply aws/database-tools/blazer.tf:48]: It doesn't and isn't and is easier than doing an import

File: aws/database-tools/cloudwatch_logs.tf [hunk 2]
```diff
@@ -1,5 +1,7 @@
...
-  name              = aws_ecs_cluster.blazer.name
-  retention_in_days = 1827 # 5 years
-}
+removed {
+  from = aws_cloudwatch_log_group.blazer
+
+  lifecycle {
+    destroy = false
+  }
+}
```
[the remark this answers is not in the document]
Comment (ben851) [reply aws/database-tools/cloudwatch_logs.tf:7]: This file will be deleted after this release so nbd

File: aws/eks/cloudwatch_log.tf [hunk 3]
```diff
@@ -2,6 +2,11 @@
 # AWS EKS Cloudwatch groups
 ###
 
+import {
```
[the remark this answers is not in the document]
Comment (ben851) [reply aws/eks/cloudwatch_log.tf:5]: Yup

File: aws/eks/cloudwatch_log.tf [hunk 4]
```diff
@@ -20,6 +25,12 @@ resource "aws_cloudwatch_log_group" "notification-canada-ca-eks-prometheus-logs"
   retention_in_days = var.log_retention_period_days
 }
 
+resource "aws_cloudwatch_log_group" "blazer" {
+  count             = var.cloudwatch_enabled ? 1 : 0
+  name              = "blazer"
+  retention_in_days = 1827 # 5 years
```
[the remark this answers is not in the document]
Comment (ben851) [reply aws/eks/cloudwatch_log.tf:31]: Yeah for some reason this one is hard coded. That can be addressed in a different card

File: aws/eks/sentinel.tf [hunk 5]
```diff
@@ -43,6 +43,7 @@ resource "aws_cloudwatch_log_subscription_filter" "admin_api_request" {
 
 resource "aws_cloudwatch_log_subscription_filter" "blazer_logging" {
   count           = var.cloudwatch_enabled ? 1 : 0
+  depends_on      = [aws_cloudwatch_log_group.blazer]
```
[the remark this answers is not in the document]
Comment (ben851) [reply aws/eks/sentinel.tf:46]: there is no log_group_name reference here - and we're referencing this as a local.

File: aws/system_status/lambda.tf [hunk 6]
```diff
@@ -25,7 +25,7 @@ module "system_status" {
     system_status_admin_url        = var.system_status_admin_url
     system_status_api_url          = var.system_status_api_url
     system_status_bucket_name      = "notification-canada-ca-${var.env}-system-status"
-    sqlalchemy_database_reader_uri = "postgresql://app_db_user:${var.app_db_user_password}@${var.database_read_only_proxy_endpoint}/NotificationCanadaCa${var.env}"
+    sqlalchemy_database_reader_uri = "postgresql://app_db_user:${var.app_db_user_password}@${var.database_read_only_proxy_endpoint}/${var.rds_database_name}"
```
[the remark this answers is not in the document]
Comment (ben851) [reply aws/system_status/lambda.tf:28]: yes

File: scripts/terragruntInitRecursive.sh [hunk 7]
```diff
@@ -18,7 +18,6 @@ runCommand()
 toApply="common,ecr,ses_receiving_emails,dns,ses_validation_dns_entries,cloudfront,eks,elasticache,rds,lambda-api,lambda-admin-pr,performance-test,heartbeat,database-tools,lambda-google-cidr,ses_to_sqs_email_callbacks,sns_to_sqs_sms_callbacks"
 
 ENVIRONMENT=$1
-VARFILE=$2
```
[the remark this answers is not in the document]
Comment (ben851) [reply scripts/terragruntInitRecursive.sh:21]: no it's not

**Review by P0NDER0SA — APPROVED** [review]
LGTM

**Pull request merged**